src/module/mod.rs

```diff
@@ -205,7 +205,9 @@ impl Module {
             .add_processed_by("walrus", env!("CARGO_PKG_VERSION"));
 
         // TODO: probably run this in a different location
-        crate::passes::validate::run(&ret)?;
+        if !ret.config.skip_validate {
+            crate::passes::validate::run(&ret)?;
```

Comment:
So we still do *most* validation, just not the stuff that happens after parsing? Seems like maybe we should rename this or at least clarify...

Reply:
Indeed yeah, although I think we'll long-term want to move all the type checking validation done during instructin parsing to the explicit validation pass if we can. In any case *some* validation happens when parsing still so I'll clarify the name here